### PR TITLE
Collect hypervisor details to SCC

### PR DESCRIPTION
**Pull request by tbazant**
### PR creator: Description

Introduces the new `scc-hypervisor-collector` tool 

### PR creator: Are there any relevant issues/feature requests?

* jsc#PED-2862

### CODE REVIEW

File: scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    <para>
+      &scchvc; is a command line tool that collects details about customer
+      hypervisor solutions and uploads them to SCC.
+    </para>
+  </sect1>
+  <sect1 xml:id="scc-hypervisor-collector-requirements">
+    <title>Requirements</title>
+
+    <para>
+      To collect hypervisor data and upload them to SCC, you need to:
```
Comment (rtamalin) [inline scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml:52]: The "tone" of this section may not "fit" with the original intent of the scc-hypervisor-collector...

The original goal was to provide a tool that customers could easily use without necessarily having to setup a special host to run it, hence the focus on providing a container image as well that can be run from any host that supports running containers with docker or podman...

So perhaps the phrasing here should be more along the lines of:

The easiest way to run scc-hypervisor-collector is on an existing registered SLE-based host...

File: scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+      </listitem>
+      <listitem>
+        <para>
+          Install the <package>scc-hypervisor-collector</package> package and
+          configure the collector to match your environment.
+        </para>
+      </listitem>
+    </itemizedlist>
+  </sect1>
+  <sect1 xml:id="scc-hypervisor-collector-how-it-works">
```
Comment (rtamalin) [inline scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml:76]: This section probably should start with some preamble describing the nature of the data being collected, e.g. the basic hardware capabilities (RAM, CPUs) of the hypervisor host (VMWare or Libvirt), the system UUID of a Libvirt host, as well as the system UUIDs of the managed VMs, and whether they are running or not. This data is used to correlate discovered Libvirt hosts and VMs with system registrations with the SCC.

File: scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        <dm:url>https://bugzilla.suse.com/enter_bug.cgi</dm:url>
+        <dm:component>Documentation</dm:component>
+        <dm:assignee>tbazant@suse.com</dm:assignee>
+      </dm:bugtracker>
+      <dm:editurl>https://github.com/SUSE/doc-unversioned/blob/main/xml/</dm:editurl>
+      <dm:translation>yes</dm:translation>
+    </dm:docmanager>
+    <abstract>
+      <para>
+        This article describes how to provide customers with details about the
```
Comment (rtamalin) [inline scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml:34]: Not sure if we want to mention here, or later, that the supported hypervisor backends are currently limited to VMWare and Libvirt

File: scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+        </para>
+      </listitem>
+    </itemizedlist>
+  </sect1>
+  <sect1 xml:id="scc-hypervisor-collector-how-it-works">
+    <title>How does &scchvc; work?</title>
+
+    <para>
+      The
+      <systemitem class="daemon">scc-hypervisor-collector.timer</systemitem>
```
Comment (rtamalin) [inline scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml:81]: I would recommend structuring this along the lines of manually configuring the command to run via the scchvc user account first, then validate that it runs correctly when triggered by the systemd service, and only when that is working should there be mention of the systemd timer.

That may mean swapping section 3 and 4?

File: scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml (hunk 3 above)
Comment (tbazant) [reply scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml:34]: I'll mention this in the requirements section

File: scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+      </para>
+      <itemizedlist>
+        <listitem>
+          <para>
+            Customer systems that have the necessary access to be able to
+            collect the data from the configured hypervisor systems may not
+            have access to the internet to be able to upload the collected data
+            to SCC. Conversely, systems that have internet access to upload
+            data to the internet may not have sufficient access to be able to
+            collect the data from the customer hypervisor solutions.
```
[the remark this answers is not in the document]
Comment (dariavladykina) [reply scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml:260]: It is unclear what scenario the separation fulfils in relation to this?

File: scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+          </para>
+        </listitem>
+        <listitem>
+          <para>
+            You want to sanitize the collected data before they are uploaded to
```
Comment (dariavladykina) [inline scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml:265]:
```suggestion
            You want to sanitize the collected data before it is uploaded to
```

**Review by dariavladykina — COMMENTED** [review]
Hi, please see some suggestions. Thanks!

File: scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+        This separation helps to fulfil the following scenarios:
+      </para>
+      <itemizedlist>
+        <listitem>
+          <para>
+            Customer systems that have the necessary access to collect the
+            data from the configured hypervisor systems may not have
+            access to the Internet to upload the collected data to SCC.
+            Conversely, systems that have Internet access to upload
+            data to the Internet may not have sufficient access to collect
```
Comment (tbazant) [inline scc-hypervisor-collector/xml/art-scc-hypervisor-collector.xml:259]:
```suggestion
            data to SCC may not have sufficient access to collect
```

**Pull request merged**